### PR TITLE
HAAR-1186 model re-factor - part 1

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

**Comment by AlTurner-MOJ** [comment]
Couple of points on my TODO list on this:

1. Rename classes in model package to remove _Details_ postfix. So `DeliusUserDetails` would become just `DeliusUser`
2. Move out `UserDetailsDto` (or rename). This class looks like it is a view model class, but need to confirm this.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/EmailDomainController.kt [hunk 1]
```diff
@@ -178,7 +179,16 @@ data class EmailDomainDto(
...
   val description: String
-)
+) {
+  companion object {
+    fun fromDomain(emailDomain: EmailDomain) =
+      EmailDomainDto(
+        emailDomain.id,
+        emailDomain.domain,
+        emailDomain.description
+      )
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/EmailDomainController.kt:189]: I would probably just do this as an extra constructor

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt [hunk 2]
```diff
@@ -133,7 +134,7 @@ class RolesController(
   @GetMapping("/roles")
   fun getRoles(
     @RequestParam(value = "adminTypes", required = false) adminTypes: List<AdminType>?,
-  ): List<Role> = rolesService.getRoles(adminTypes)
+  ): List<RoleDto> = rolesService.getRoles(adminTypes).map { RoleDto.fromDomain(it) }
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt:137]: same with this one

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt [hunk 3]
```diff
@@ -356,16 +357,16 @@ data class Role(
...
-    r.roleName,
-    r.roleDescription,
-    r.adminType
-  )
+  companion object {
+    fun fromDomain(role: Role): RoleDto =
+      with(role) {
+        return RoleDto(roleCode, roleName, roleDescription, adminType)
+      }
+  }
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt:365]: I think this should stay as a constructor, passing in the Role to make the RoleDto

**Review by karenmillermoj — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/EmailDomainController.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/EmailDomainController.kt:189]: Discussed with @karenmillermoj and agreed OK

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt:137]: Discussed with @karenmillermoj and agreed OK

**Pull request merged**